File: app/src/androidTest/java/com/pleon/buyt/DatabaseTest.java
package com.pleon.buyt;

import android.content.Context;

import com.pleon.buyt.database.AppDatabase;
import com.pleon.buyt.database.dao.ItemDao;
import com.pleon.buyt.model.Item;

import org.junit.After;
import org.junit.Before;
import org.junit.runner.RunWith;


import androidx.room.Room;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.platform.app.InstrumentationRegistry;

import static org.junit.Assert.*;

// Instrumented test, which will execute on an Android device.
@RunWith(AndroidJUnit4.class)
public class DatabaseTest {

    private AppDatabase database;
    private ItemDao itemDao;

    @Before
    public void setUp() {
        // Context of the app under test.
        Context context = InstrumentationRegistry.getInstrumentation().getContext();
        database = Room.inMemoryDatabaseBuilder(context, AppDatabase.class).build();
        itemDao = database.itemDao();
    }

    public void addOneItem() {
        Item item = new Item("Chocolate", "1225", 1, "None");

        itemDao.insert(item);

        long itemCount = itemDao.getCount();
        assertEquals(1, itemCount);
    }

    @After
    public void tearDown() {
        database.close();
    }
}

File: app/src/androidTest/kotlin/com.pleon.buyt/DatabaseTest.kt
package com.pleon.buyt


import androidx.room.Room
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.platform.app.InstrumentationRegistry
import com.pleon.buyt.database.AppDatabase
import com.pleon.buyt.database.dao.ItemDao
import com.pleon.buyt.model.Item
import org.junit.After
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.runner.RunWith

// Instrumented test, which will execute on an Android device.
@RunWith(AndroidJUnit4::class)
class DatabaseTest {

    private var database: AppDatabase? = null
    private var itemDao: ItemDao? = null

    @Before
    fun setUp() {
        // Context of the app under test.
        val context = InstrumentationRegistry.getInstrumentation().context
        database = Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java).build()
        itemDao = database!!.itemDao()
    }

    fun addOneItem() {
        val item = Item("Chocolate", "1225", 1, "None")

        itemDao!!.insert(item)

        val itemCount = itemDao!!.getCount()
        assertEquals(1, itemCount)
    }

    @After
    fun tearDown() {
        database!!.close()
    }
}
